pychromecast/socket_client.py

```diff
@@ -403,7 +403,11 @@ def run_once(self):
                 function = callback['function']
                 event.set()
                 if function:
-                    function(data)
+                    try:
+                        function(data)
+                    except Exception as e:
+                        self.logger.error("runonce: Error calling callback  : %s" % str(e))
+                        # maybe do something else...
```

Comment:
Should we close the connection and return 1?

Reply:
It actually depends on for what kind of action the callback is required. It depends of the code in the callback function.

In my case it was when launching an app and error is on google cast device side so it should not close the connection (so return 0). It happened on some case when an application is locked on the cast or if app id is not found.
The 'new_launch_error' listener should implement the logic to handle the correct behavior when it happens but this is not a socket issue.

Maybe in some other cases, the cause is more serious and could require disconnection but I'm not really an expert in this.
My guess would be to never disconnect.

I also believe that this could explained some previous cases I experienced when the callback failed (for whatever reason) and was causing loss of control with pychromecast. 
It could be judicious to show in the logs the underlying/parent exception for debug purpose. The `self.logger.exception('runonce: error calling callback')` will not give enough information.

An alternative way would be remove the exception raised on this part as it's not really relevant in that case (it's just a regular error status and not an exception).
```
# original
def _block_till_launched(self, app_id):
        if self.blocking:
            self.app_launch_event.wait()
            if self.launch_failure:
                raise LaunchError(
                    "Failed to launch app: {}, Reason: {}".format(
                        app_id, self.launch_failure.reason))

# just logging instead of raising exception
def _block_till_launched(self, app_id):
        if self.blocking:
            self.app_launch_event.wait()
            if self.launch_failure:
                self.logger.error(
                    "Failed to launch app: {}, Reason: {}".format(
                        app_id, self.launch_failure.reason))
```

It's kind of up to you.

Reply:
`self.logger.exception` prints the full stack trace of all exceptions, it will give all the needed information.

I like failing hard. So if the app that is using Chromecast is not handling something correctly, we should blow up . That way the developer knows their app is not working as expected.